packages/parser-babel-config/src/babel-config.ts

```diff
@@ -0,0 +1,109 @@
+import { Sonarwhal } from 'sonarwhal';
+import { loadJSONFile } from 'sonarwhal/dist/src/lib/utils/misc';
+
+import { BabelConfig, BabelConfigInvalid, BabelConfigParsed, BabelConfigInvalidSchema } from './BabelConfigParse';
+
+export default class BabelConfigParser extends Parser {
+    private configFound: boolean = false;
+    private schema: any;
+    private validator: ajv.Ajv;
+
+    public constructor(sonarwhal: Sonarwhal) {
+        super(sonarwhal);
+        this.schema = loadJSONFile(path.join(__dirname, 'schema', 'babelConfigSchema.json'));
+
+        /*
+         * If we want to use the ajv types in TypeScript, we need to import
+         * ajv in a lowsercase variable 'ajv', otherwise, we can't use types
+         * like `ajv.Ajv'.
+         */
+        this.validator = new ajv({ // eslint-disable-line new-cap
+            $data: true,
+            allErrors: true,
+            schemaId: 'id',
+            verbose: true
+        });
+
+        this.validator.addMetaSchema(require('ajv/lib/refs/json-schema-draft-04.json'))
+
+        /**
+         * .babelrc => type: 'unknown' ('file-type' module doesn't support the type of 'json').
+         * package.json => type: 'json' (file type from extention).
+         */
+        sonarwhal.on('fetch::end::json', this.parseBabelConfig.bind(this));
+        sonarwhal.on('fetch::end::unknown', this.parseBabelConfig.bind(this));
+        sonarwhal.on('scan::end', this.parseEnd.bind(this));
+    }
+
+    private async parseEnd(scanEnd: ScanEnd) {
+        if (!this.configFound) {
+            await this.sonarwhal.emitAsync('notfound::babel-config', scanEnd);
+        }
+    }
+
+    private async validateSchema(config: BabelConfig, resource: string) {
+        // ajv is lower case to be able to get the types.
+        const validate: ajv.ValidateFunction = this.validator.compile(this.schema);
+        const valid = validate(config);
+
+        if (!valid) {
+            const event: BabelConfigInvalidSchema = {
+                errors: validate.errors,
+                resource
+            };
+
+            await this.sonarwhal.emitAsync('invalid-schema::babel-config', event);
+        }
+
+        return valid;
+    }
+
+    private async parseBabelConfig(fetchEnd: FetchEnd) {
+        const resource = fetchEnd.resource;
+        const allowedFileNames = ['.babelrc', 'package.json'];
+        const resourceFileName = path.basename(resource);
+        const isPackageJson: boolean = resourceFileName === 'package.json';
+        const isBabelrc: boolean = resourceFileName === '.babelrc';
+
+        if (!isBabelrc && !isPackageJson) {
+            return;
+        }
+
+        this.configFound = true;
+        let config: BabelConfig;
+
+        try {
+            const response = fetchEnd.response;
+            // When using local connector to read local files, 'content' is empty.
+            const content = JSON.parse(response.body.content || response.body.rawContent.toString());
```

Comment:
We shouldn't need to do this. The response is a `string` so there should be something on `content` and in the TypeScript one we don't need it. Can you investigate a bit more under what circumstances we need to access the `rawContent`?

Reply:
@molant I investigated in the `local` connector, and this is due to the `unknown` media type too. With the media type of `.babelrc` as `unknown`, the following lines don't run, so `content` is empty. https://github.com/sonarwhal/sonarwhal/blob/43f395fcfe0bca655dcb1013f07a386811214191/packages/sonarwhal/src/lib/connectors/local/local.ts#L89-L91